wire-ios-data-model/Source/ConversationSearch/FilterConversationsUseCase.swift

```diff
@@ -0,0 +1,74 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import WireUtilities
+
+/// This use case takes grouped conversations as input and returns the same groups with their conversations
+/// filtered by the search text `query`.
+///
+/// The search text is compared to the name of conversations as well as the names of the conversation's participants.
+public struct FilterConversationsUseCase<ConversationContainer>: FilterConversationsUseCaseProtocol
```

Comment:
Just sharing a thought - not a request for change... 

Part of me wonders if this should be done at the core data layer (if that is even possible). I guess by doing it here we can have more flexibility.

Reply:
Good question, I don't know. I have little experience with search in general. The way of searching is also not flexible, it just calls String.contains.
The other point why I did this isolated was because the code for showing the conversation list is really hard to maintain. I hope we can refactor the whole feature (conversation list including search) eventually.